### PR TITLE
feat: migration of demo to svelte 5

### LINKED ISSUE #944: Bug: demo : "On this Page" is not refreshed when changing a route client side
Svelte 5 regression
On this page menu does not change anymore when navigating client side.

### PR DESCRIPTION
**Pull request by quentinderoubaix**
closes #944

### CODE REVIEW

File: demo/src/lib/layout/ComponentTypeAlert.svelte [hunk 1]
```diff
@@ -4,23 +4,28 @@
...
-	$: componentName = $page.url.pathname.match(regex)?.[2];
-	$: {
+	let type = $derived('info' as AlertProps['type']);
+	let componentName = $derived($page.url.pathname.match(regex)?.[2]);
+	$effect(() => {
 		// eslint-disable-next-line @typescript-eslint/no-unused-expressions
 		componentName;
-		alert?.api.open();
-	}
+		untrack(() => alert!.api.open());
```
Comment (divdavem) [inline demo/src/lib/layout/ComponentTypeAlert.svelte:27]: Please add a comment explaining why this workaround of using `untrack` is needed, with the reference to the (existing or new :wink:) Svelte issue that requests the fix of this problem in Svelte.

File: demo/src/lib/layout/StatusAlert.svelte [hunk 2]
```diff
@@ -4,29 +4,31 @@
...
+			: 'This component has all the basic functionalities implemented. More polishing features might be added in the near future.',
+	);
+	let componentName = $derived($page.url.pathname.match(regex)?.[2]);
+	let issueUrl = $derived(`https://github.com/AmadeusITGroup/AgnosUI/issues?q=is%3Aopen+is%3Aissue+label%3A%22widget%3A+${componentName}%22`);
+	$effect(() => {
 		// eslint-disable-next-line @typescript-eslint/no-unused-expressions
 		componentName;
-		alert?.api.open();
-	}
+		untrack(() => alert!.api.open());
```
Comment (divdavem) [inline demo/src/lib/layout/StatusAlert.svelte:30]: (same comment as for the other untrack usage)

File: demo/src/lib/layout/ComponentTypeAlert.svelte (hunk 1 above)
Comment (quentinderoubaix) [reply demo/src/lib/layout/ComponentTypeAlert.svelte:22]: Comment added and issue on our side created !

File: demo/src/lib/layout/StatusAlert.svelte (hunk 2 above)
Comment (quentinderoubaix) [reply demo/src/lib/layout/StatusAlert.svelte:31]: Comment added and issue on our side created !

**Review by divdavem — APPROVED** [review]

**Pull request merged**